File: src/ReetCode/Easy/1 ~ 100/Q35/Q35_SearchInsertPosition.java
/**
* Author    : Lee In Bok 
* Date      : 2023.11.01(Wed)
* Spend Time: 04m 40s
* Runtime   : 0 ms (Beats 100%)
* Memory    : 43.43 MB (Beats 22.47%)
* Algoritm  : Array
 */

package Q35;

public class Q35_SearchInsertPosition {
    public static void main(String[] args){
        int result = searchInsert(new int[]{1, 3, 5, 6}, 5);

        System.out.println(result);
    }

    public static int searchInsert(int[] nums, int target) {
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == target || nums[i] > target){
                return i;
            }
        }

        return nums.length;
    }
}

File: src/ReetCode/Easy/1 ~ 100/Q35/Q35_Search Insert Position.py
"""
# Author    : Lee In Bok 
# Date      : 2023.11.01(Wed)
# Spend Time: 04m 40s
# Runtime   : 50 ms (Beats 80.77%)
# Memory    : 16.9 MB (Beats 89.35%)
# Algoritm  : Array
"""

from typing import List

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        for idx in range(len(nums)):
            if nums[idx] == target or target < nums[idx]:
                return idx
            elif idx == len(nums) - 1:
                return len(nums)

s = Solution()
result = s.searchInsert([1,3,5,6], 7)

print(result)